### PR TITLE
TAJO-2052: Upgrading ORC reader version

### PR DESCRIPTION
**Pull request by eminency**
Code is refined, and it is based on presto-orc-0.132.

* Base data structure is changed from Vector to Block.
* Some bugs are fixed.
* Compatibility with Hive is improved.
* Unluckily, there is no speed improvement.

### CODE REVIEW

**Comment by blrunner** [comment]
@eminency 

Thank you for your contribution.
When using MySQLStore, this PR runs successfully. But when using HiveCatalogStore, it throws NoClassDefFoundError as following:

```
2016-01-28 13:25:53,025 ERROR org.apache.tajo.master.GlobalEngine: 
Stack Trace:
java.lang.NoClassDefFoundError: com/facebook/presto/hive/protobuf/CodedInputStream
	at com.facebook.presto.orc.metadata.OrcMetadataReader.readPostScript(OrcMetadataReader.java:48)
	at com.facebook.presto.orc.OrcReader.<init>(OrcReader.java:99)
	at org.apache.tajo.storage.orc.ORCScanner.init(ORCScanner.java:136)
	at org.apache.tajo.engine.planner.physical.SeqScanExec.initScanner(SeqScanExec.java:286)
	at org.apache.tajo.engine.planner.physical.SeqScanExec.init(SeqScanExec.java:191)
	at org.apache.tajo.engine.planner.physical.PartitionMergeScanExec.initScanExecutors(PartitionMergeScanExec.java:80)
	at org.apache.tajo.engine.planner.physical.PartitionMergeScanExec.init(PartitionMergeScanExec.java:67)
```

For the reference, I added the installed directory of apache hive 1.2.1 to tajo-env.sh file.

**Comment by eminency** [comment]
@blrunner 
It's already reported issue. Refer this:

https://issues.apache.org/jira/browse/TAJO-1929

I will fix it after this PR because it needs ORC upgrade.

File: tajo-storage/tajo-storage-hdfs/src/main/java/org/apache/tajo/storage/thirdparty/orc/HdfsOrcDataSource.java [hunk 1]
```diff
@@ -114,9 +117,9 @@ public void readFully(long position, byte[] buffer, int bufferOffset, int buffer
       buffers.put(mergedRange, buffer);
     }
 
-    ImmutableMap.Builder<K, Slice> slices = ImmutableMap.builder();
+    ImmutableMap.Builder<K, FixedLengthSliceInput> slices = ImmutableMap.builder();
     for (Entry<K, DiskRange> entry : diskRanges.entrySet()) {
```
Comment (blrunner) [inline tajo-storage/tajo-storage-hdfs/src/main/java/org/apache/tajo/storage/thirdparty/orc/HdfsOrcDataSource.java]: If you use lambda expression, following codes would be simple.

**Comment by blrunner** [comment]
Ok, I understood your comments. Could you check my trivial comment?

**Comment by eminency** [comment]
@blrunner 
I fixed it. Please verify if it fits in your intention.

**Comment by blrunner** [comment]
+1

LGTM! I'll commit this PR soon. :)

**Pull request closed without merging**